api/src/search/backend/load_agencies_to_index.py

```diff
@@ -73,14 +74,27 @@ def fetch_agencies(self) -> Sequence[Agency]:
 
     def load_agencies(self, agencies: Sequence[Agency]) -> None:
         logger.info("Loading agencies...")
+        active_agency_subquery = (
+            _construct_active_inner_query(Agency.agency_id)
+            .union(_construct_active_inner_query(Agency.top_level_agency_id))
+            .subquery()
+        )
+
+        agency_id_stmt = select(active_agency_subquery).distinct()
+        active_agencies = set(self.db_session.execute(agency_id_stmt).unique().scalars().all())
 
         agencies_json = []
         for agency in agencies:
             logger.info(
                 "Preparing agency for upload to search index",
                 extra={"agency_id": agency.agency_id, "agency_code": agency.agency_code},
             )
-            agencies_json.append(SCHEMA.dump(agency))
+            agency_json = SCHEMA.dump(agency)
+            if agency.agency_id in active_agencies:
+                agency_json["is_active_agency"] = True
```

Comment:
Can also set it when it's false just for completeness / one less line
```suggestion
            agency_json["is_active_agency"] = agency.agency_id in active_agencies
```